common.mk

```diff
@@ -33,6 +35,72 @@ endef
 DEBUG ?= 0
 
 ################################################################################
+# Busybox
+################################################################################
+BUSYBOX_COMMON_TARGET		?= TOBEDEFINED
+BUSYBOX_CLEAN_COMMON_TARGET	?= TOBEDEFINED
+BUSYBOX_COMMON_CCDIR		?= TOBEDEFINED
+
+busybox-common: linux
+	cd $(GEN_ROOTFS_PATH) &&  \
+		CC_DIR=$(BUSYBOX_COMMON_CCDIR) \
+		PATH=${PATH}:$(LINUX_PATH)/usr \
+		$(GEN_ROOTFS_PATH)/generate-cpio-rootfs.sh \
+			$(BUSYBOX_COMMON_TARGET)
+
+busybox-clean-common:
+	cd $(GEN_ROOTFS_PATH) && \
+	$(GEN_ROOTFS_PATH)/generate-cpio-rootfs.sh  \
+		$(BUSYBOX_CLEAN_COMMON_TARGET)
+
+busybox-cleaner-common:
+	rm -rf $(GEN_ROOTFS_PATH)/build
+	rm -rf $(GEN_ROOTFS_PATH)/filelist-final.txt
+	
+################################################################################
+# Linux
+################################################################################
+LINUX_COMMON_FLAGS ?= LOCALVERSION= CROSS_COMPILE=$(CROSS_COMPILE_NS_KERNEL)
+
+linux-common: linux-defconfig
+	$(MAKE) -C $(LINUX_PATH) $(LINUX_COMMON_FLAGS)
+
+linux-defconfig-clean-common:
+	@if [ -f "$(LINUX_PATH)/.config" ]; then \
+		rm $(LINUX_PATH)/.config; \
+	fi
+
+LINUX_CLEAN_COMMON_FLAGS ?= $(LINUX_COMMON_FLAGS)
+	
+linux-clean-common: linux-defconfig-clean
+	$(MAKE) -C $(LINUX_PATH) $(LINUX_CLEAN_COMMON_FLAGS) clean
+
+LINUX_CLEANER_COMMON_FLAGS ?= $(LINUX_COMMON_FLAGS)
+
+linux-cleaner-common: linux-defconfig-clean
+	$(MAKE) -C $(LINUX_PATH) $(LINUX_CLEAN_COMMON_FLAGS) distclean
+
+
+################################################################################
+# EDK2 / Tianocore
+################################################################################
+# Make sure edksetup.sh only will be called once and that we don't rebuild
+# BaseTools again and again.
+$(EDK2_PATH)/Conf/target.txt:
+	set -e && cd $(EDK2_PATH) && $(BASH) edksetup.sh && \
+	$(MAKE) -j1 -C $(EDK2_PATH)/BaseTools
```

Comment:
No, this will not work as variables set in `edksetup.sh` will be forgotten in the following `$(MAKE)`